easybuild/tools/toolchain/mpi.py

```diff
@@ -168,40 +168,67 @@ def mpi_family(self):
 
     def mpi_cmd_for(self, cmd, nr_ranks):
         """Construct an MPI command for the given command and number of ranks."""
-
-        # parameter values for mpirun command
         params = {
             'nr_ranks': nr_ranks,
             'cmd': cmd,
         }
-
         mpi_cmd_template = build_option('mpi_cmd_template')
         if mpi_cmd_template:
-            self.log.info("Using specified template for MPI commands: %s", mpi_cmd_template)
+            mpi_exec_template = build_option('mpi_exec_template')
+            if mpi_exec_template:
+                self.log.warning(
+                    "Overriding 'mpi_exec_template':\n%s\nwith specified 'mpi_cmd_template' for MPI commands:\n%s",
+                    mpi_exec_template,
+                    mpi_cmd_template
+                )
+            else:
+                self.log.info("Using specified template for MPI commands: %s", mpi_cmd_template)
+        else:
+            params['mpi_exec_nranks'] = self.mpi_exec_nranks(nr_ranks=nr_ranks)
+            mpi_cmd_template = "%(mpi_exec_nranks)s %(cmd)s"
+
+        try:
+            res = mpi_cmd_template % params
+        except KeyError as err:
+            raise EasyBuildError("Failed to complete MPI cmd template '%s' with %s: %s", mpi_cmd_template, params, err)
```

Comment:
It's probably better if we explicitly check that `%(nr_ranks)s` and %(cmd)s` are both included in the `mpi_cmd_template` value?

```python
missing = []
for key in params:
    tmpl = '%(' + key + ')s'
    if tmpl not in mpi_cmd_template:
        missing.append(tmpl)
if missing:
    raise EasyBuildError("Missing templates in mpi-cmd-template value '%s': %s", mpi_cmd_template, ', '.join(missing))